### ISSUE TITLE
Improved error trapping for CCD/ReadoutMode/ReadoutMode

### ISSUE THREAD

**Issue by JorisDeRidder** [IvS-KULeuven/PlatoSim3#302]
If the CCD/ReadoutMode/ReadoutMode is not either "Nominal" or "Partial" (e.g. because of an accidental typo in the inputfile.yaml), the code will silently fail.

For example in the method Simulation::configureReadoutTime() in such case the if-statements will fail, and the variable 'numRowsReadout' will be used uninitialized.

**Comment by SaraRegibo** [comment]
I tested with `CCD/ReadoutMode/ReadoutMode` = `Test` (arbitrary choice).  In that case, the simulations don't even start, but you get the following error message:

```
2018-10-05 11:56:32 ERROR   Detector::configure(): Unkown readout mode specification in configuration file: Test
2018-10-05 11:56:32 ERROR   Detector: Unkown readout mode specification in configuration file
libc++abi.dylib: terminating with uncaught exception of type ConfigurationException: Detector: Unkown readout mode specification in configuration file
Abort trap: 6
```

So it does not silently fail, but you get an error message that points you immediately to the problem.

**Comment by JorisDeRidder** [comment]
Detector::configure() is ran after Simulation::configureReadoutTime() so the latter has already been silently failing by that time. I wonder if it's a good idea to move the error trapping to the earliest place where we rely on the value of ReadoutMode (which would be in Simulation).

But it's good that the user is already made aware, so this issue has IMO a lower priority.

**Comment by SaraRegibo** [comment]
Solved in pull request #324.  Assigning to @JorisDeRidder for closure.

**Comment by SaraRegibo** [comment]
Closed by pull request #324
